local-libs/traceviewer-libs/react-components/src/components/generic-xy-output-component.tsx

```diff
@@ -710,6 +726,109 @@ export class GenericXYOutputComponent extends AbstractTreeOutputComponent<Generi
         }
     }
 
+    private renderTimeline(): React.ReactNode {
+        if (this.state.timelineUnitType === 'time' && !this.isTimeAxis) return <svg/>;
+        
+        const chartWidth = this.getChartWidth();
+        if (chartWidth <= 0) return <svg/>;
+
+        return (
+            <svg 
+                ref={this.timelineRef}
+                width={chartWidth} 
+                height={this.timelineHeight}
+                style={{ marginLeft: this.margin.left }}
+            />
+        );
+    }
+
+    private updateTimeline(): void {
+        if (!this.timelineRef.current) return;
+        if (this.state.timelineUnitType === 'time' && !this.isTimeAxis) return;
```

Comment:
The timelineUnitType is always 'time', even for ranges as x-axis or categories a x-axis. For ranges or categories it will return here because for those xy series `this.isTimeAxis` is false.